mountpoint-s3/src/fs.rs

```diff
@@ -58,8 +58,39 @@ where
 {
     inode: Inode,
     full_key: String,
-    object_size: u64,
-    typ: FileHandleType<Client, Prefetcher>,
+    state: FileHandleState<Client, Prefetcher>,
```

Comment:
Doesn't this have to be inside a mutex? Or we'll get a race condition when re-inserting a new value?